server/src/services/eliza.service.ts

```diff
@@ -398,44 +405,46 @@ export class MessageManager {
       const content: Content = {
         text: fullText,
         source: "telegram",
-        // inReplyTo:
-        //     "reply_to_message" in message && message.reply_to_message
-        //         ? stringToUuid(
-        //               message.reply_to_message.message_id.toString() +
-        //                   "-" +
-        //                   this.runtime.agentId
-        //           )
-        //         : undefined,
+        inReplyTo:
+          "reply_to_message" in message && message.reply_to_message
+            ? stringToUuid(
+                message.reply_to_message.message_id.toString() +
+                  "-" +
+                  this.runtime.agentId
+              )
+            : undefined,
       };
 
       // Create memory for the message
-      const memory: Memory = {
+
+      const memory = await this.runtime.messageManager.addEmbeddingToMemory({
         id: messageId,
         agentId,
         userId,
         roomId,
         content,
         createdAt: message.date * 1000,
-        embedding: getEmbeddingZeroVector(),
-      };
-
-      await this.runtime.messageManager.createMemory(memory);
+      });
+      // set unique to avoid duplicating memories
+      await this.runtime.messageManager.createMemory(memory, true);
```

Comment:
callout: `unique=true`

Reply:
seems to be the default internally even though that's not clear in the docs 🤷

Reply:
ah, this was directly forked from the `telegram-client` service of the eliza repo
need to see if they added comments there or not
but that's one of the reasons we wanted to start the starter kit!